The Windows failure in the PYTHONSTARTUP test is a bit peculiar. It's actually an exception in asyncio proper:

```
Traceback (most recent call last):
  File "C:\Users\ambv\OneDrive\Documents\GitHub\cpython\Lib\runpy.py", line 189, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
                               ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "C:\Users\ambv\OneDrive\Documents\GitHub\cpython\Lib\runpy.py", line 148, in _get_module_details
    return _get_module_details(pkg_main_name, error)
  File "C:\Users\ambv\OneDrive\Documents\GitHub\cpython\Lib\runpy.py", line 112, in _get_module_details
    __import__(pkg_name)
    ~~~~~~~~~~^^^^^^^^^^
  File "C:\Users\ambv\OneDrive\Documents\GitHub\cpython\Lib\asyncio\__init__.py", line 43, in <module>
    from .windows_events import *
  File "C:\Users\ambv\OneDrive\Documents\GitHub\cpython\Lib\asyncio\windows_events.py", line 8, in <module>
    import _overlapped
OSError: [WinError 10106] The requested service provider could not be loaded or initialized
```

It's definitely related though, so I'm investigating.

--- a/Lib/test/test_repl.py
+++ b/Lib/test/test_repl.py
@@ -1,9 +1,6 @@
 """Test the interactive interpreter."""
 
-from io import BytesIO
 import os
-from re import sub
-import tempfile
 import select
 import subprocess
 import sys
@@ -13,10 +10,11 @@ from test import support
 from test.support import (
     cpython_only,
     has_subprocess_support,
+    os_helper,
     SuppressCrashReport,
     SHORT_TIMEOUT,
 )
-from test.support.script_helper import assert_python_failure, kill_python, assert_python_ok
+from test.support.script_helper import kill_python
 from test.support.import_helper import import_module
 
 try:
@@ -210,16 +208,20 @@ class TestInteractiveInterpreter(unittest.TestCase):
         self.assertIn(expected, output, expected)
 
     def test_asyncio_repl_reaches_python_startup_script(self):
-        with tempfile.TemporaryDirectory() as tmpdir:
-            script = os.path.join(tmpdir, "script.py")
+        with os_helper.temp_dir() as tmpdir:
+            script = os.path.join(tmpdir, "pythonstartup.py")
             with open(script, "w") as f:
-                f.write("exit(0)")
+                f.write("print('pythonstartup done!')" + os.linesep)
+                f.write("exit(0)" + os.linesep)
 
+            env = os.environ.copy()
+            env["PYTHONSTARTUP"] = script
             subprocess.check_call(
                 [sys.executable, "-m", "asyncio"],
                 stdout=subprocess.PIPE,
                 stderr=subprocess.PIPE,
-                env={"PYTHONSTARTUP": script}
+                env=env,
+                timeout=SHORT_TIMEOUT,
             )
 
     @unittest.skipUnless(pty, "requires pty")

